### PR TITLE
Use eslint-local-rules plugin

### PR DESCRIPTION
**Pull request by fredericgermain**
This allows to avoid having errors in editors that
supports eslint

cf https://github.com/eslint/eslint/issues/8769

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | `Fixes #1, Fixes #2` 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

### CODE REVIEW

File: eslint-local-rules.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// @flow
```
Comment (loganfsmyth) [inline eslint-local-rules.js]: Is there any way to avoid putting this in the repo root?

**Comment by mAAdhaTTah** [comment]
This would def benefit Atom. I know I've seen those errors and just ignored them up to this point.

**Comment by fredericgermain** [comment]
It fixes the warnings in Atom.
Probably it fixes in VSCode without having to set the rules path manually.

I reworked the commit to remove the flow line.

**Comment by loganfsmyth** [comment]
Manually merged in https://github.com/babel/babel/commit/45c28fa78573808e3891a13f7a6fff9bf33676d9.

**Pull request closed without merging**